Feat: 게시물 일반/랭킹 조회 페이지 구현 [FRONT-179]

sidsidsidsids opened the pull request:
### Changes
게시물 일반/랭킹 조회 페이지 구현
atoms/organisms 생성 및 수정
api - posts.api 생성

여러 프로필 간 게시물 조회 시 감정표현 변화 테스트, 게시판 페이지네이션 테스트를 따로 진행하지 않았습니다
프로젝트 mvp 페이지들을 구성하여 페이지 간 연결 이후 다시 테스트하도록 하겠습니다 
<br>

### Test Checklist ☑️
- [x] 코드에 불필요한 부분이 있는지 확인해주세요
- [x] 게시물 조회/감정표현 로직에서 고려하지 않은 부분이나 염려되는 점이 있는지 확인해주세요

### Screenshot(option)
![FRONT-179(1)](https://github.com/BbeumbungE/Frontend/assets/122246882/0d3b3047-c690-48cf-bba2-d57111ade15d)
![FRONT-179(2)](https://github.com/BbeumbungE/Frontend/assets/122246882/daa97830-c967-46ac-8988-f0ad9fb18284)
![FRONT-179(3)](https://github.com/BbeumbungE/Frontend/assets/122246882/5b8ad876-3e93-459f-9d12-10e427fda6df)
![FRONT-179(4)](https://github.com/BbeumbungE/Frontend/assets/122246882/b4708fe8-acaa-46c9-9ab3-b917914d2674)

<br>

sidsidsidsids committed:
Feat: 랭킹 게시물 조회 페이지 components 생성 [FRONT-179]
- atoms/organisms 생성
- api 생성 (posts.ts)
- RankingPage 진행중
Related to: issue #179

sidsidsidsids committed:
Merge branch 'fe/dev' of https://github.com/BbeumbungE/Frontend into feat/FRONT-179

sidsidsidsids committed:
Feat: PageHeaderText props 수정 및 그에 따른 연동 페이지 수정 [FRONT-179]
- fontsize props 추가
Related to: issue #179

sidsidsidsids committed:
Feat: 게시물 조회 페이지 atoms/organisms 구현 [FRONT-179]
- DetailRankPostBox 생성
- 나머지는 수정
Related to: issue #179

sidsidsidsids committed:
Feat: Page 초기 템플릿 구현 [FRONT-179]
- api 초기 구현
- 페이지들 초기 구현
Related to: issue #179

sidsidsidsids committed:
Feat: 블러 박스 상태 나가기 oragnism 구현 [FRONT-179]
- onClick을 props로 받는 나가기 버튼
Related to: issue #179

sidsidsidsids committed:
Design: Components CSS 수정 [FRONT-179]
- margin 수정 및 필요 없는 props 제거
Related to: issue #179

sidsidsidsids committed:
Feat: 페이지 API 연결 및 구현 [FRONT-179]
- 초기 페이지 및 상세 조회 구현
Related to: issue #179

sidsidsidsids committed:
Merge branch 'fe/dev' of https://github.com/BbeumbungE/Frontend into feat/FRONT-179

sidsidsidsids committed:
Fix: 불필요 코드 삭제 [FRONT-179]
-BlurBox onClick 만들었다가 다 못지운거 마저 지움
Related to: issue #179

YeryunJung commented on client/src/pages/RankingPage.tsx:
```diff
@@ -1,8 +1,253 @@
+  const [posts, setPosts] = useState<any>([]);
+  const [subjectTitle, setSubjectTitle] = useState<string>('');
+  const userProfile = useRecoilValue(UserProfileState);
+  // 게시물 상세 state
```
감정별로 상태가 개별로 만들어져 있어 코드가 길어질 수도 있고 추후 감정이 추가될 가능성을 고려해
감정 변수에 관련된 변수는 하나의 객체로 그룹화하여 useState 로 관리하는 것이 어떨까 추천드립니다 

관련하여 찾아보니
객체로 상태를 관리할 경우, 개별 상태 변수를 직접 갱신하는 것이 아니라 객체 전체를 갱신해 불필요한 렌더링을 유발할 수 있다고 하는데 
저희 모달에서 감정개수가 출력되는 부분이 큰 영역을 차지하고 있지 않아 크게 고려하지 않아도 괜찮을 것 같습니다!

sidsidsidsids replied:
해당 부분 작성하면서도 코드가 길어진다 생각했는데 좋은 방법이라 생각됩니다.
수정하겠습니다!

YeryunJung commented on client/src/pages/RankingPage.tsx:
```diff
@@ -1,8 +1,253 @@
+      {posts.rankerList && (
+        <PostContainer>
+          {posts.rankerList.map((post: any, index: number) => (
+            <RankPicturePostDiv
```
map으로 순회하여 렌더링하는 경우에
key 값에 index 를 사용할 경우 추후 배열이 변경되었을 때 key 값이 바뀌게 되고 key를 기반으로 하는 메모이제이션 기능의 의미가 사라져 성능 저하의 원인이 될 수 있다고 합니다
저 같은 경우 데이터로 같이 전달오는 고유 id 값을 키 값으로 많이 쓰는 편인데 이 부분 고려해주시면 코드 품질이 더더 높아질 것 같습니다!! ☺

sidsidsidsids replied:
지금 현재 받는 rank 데이터가 리스트로 받아 각 원소는 {canvasId, canvasUrl, postId, profileId} 를 가지고 있습니다.
postId로 key를 설정하면 게시물을 먼저 생성한 순서대로 출력을 할 거 같아서 rank 순서대로 출력을 해야한다는 생각으로 index로 임의의 key로 설정하여 사용했는데.. 제가 key를 잘못 이해하고 있나요? 다른 방법이 있을까요?

YeryunJung replied:
구두로 말씀드린대로 키는 고유의 값을 가지면 되고 출력순서에는 영향이 없기 때문에 posiId를 하는 것도 좋은 방법으로 보입니다!

YeryunJung commented:
여러 기능이 한 UI에 포함되어있어서 구현이 복잡하셨을텐데
가독성좋고 깔끔하게 작성해주신 것으로 보입니다
특히 그림 요소를 하드코딩 없이 효율적으로 순회해서 출력하신 것이 인상깊었습니다! 

코멘트 확인해주시고 넘넘 고생많으셨습니다아

YeryunJung reviewed (commented):
여러 기능이 한 UI에 포함되어있어서 구현이 복잡하셨을텐데
가독성좋고 깔끔하게 작성해주신 것으로 보입니다
특히 그림 요소를 하드코딩 없이 효율적으로 순회해서 출력하신 것이 인상깊었습니다!

코멘트 확인해주시고 넘넘 고생많으셨습니다아

sidsidsidsids committed:
Feat: update 감정표현 추가 [FRONT-179]
- updateEmotion 로직 추가 및 게시물 조회 횟수 단축 위한 로직 수정
- legacy로 입력된 고정값 수정
Related to: issue #179
